implementations/elixir/ockam/ockam_typed_cbor/lib/typed_cbor/plugin.ex

```diff
@@ -54,6 +50,24 @@ defmodule Ockam.TypedCBOR.Plugin do
 
   defp field_schema(schema, _), do: %{schema: schema, required: true}
 
+  defp validate_key_sequence(mod, :list) do
+    unless sequential_keys?(mod) do
+      raise("keys must be a sequence of integers starting at 1")
+    end
+  end
+
+  defp validate_key_sequence(mod, :struct) do
+    unless sequential_keys?(mod) do
+      IO.warn("keys should be a sequence of integers starting at 1")
+    end
+  end
+
+  defp sequential_keys?(mod) do
+    fields = Module.get_attribute(mod, :tt_fields)
+    keys = Enum.map(fields, fn {_, f} -> f[:key] end)
+    Enum.sort(keys) == Enum.sort(1..Enum.count(keys))
```

Comment:
This is likely too much, fields can start at 0 or 1 as it is currently 🤷‍♂️.  But, I'm 🟢  to merge,  we can address this once we switch to this encoding and start testing the rust <> elixir interoperability.